lib/pbench/test/unit/server/test_schema.py

```diff
@@ -83,25 +85,32 @@ class TestParameter:
     def test_constructor(self):
         x = Parameter("test", ParamType.STRING)
         assert not x.required
+        assert not x.nullable
         assert x.name == "test"
         assert x.type is ParamType.STRING
 
         y = Parameter("foo", ParamType.JSON, required=True)
         assert y.required
+        assert not y.nullable
         assert y.name == "foo"
         assert y.type is ParamType.JSON
 
+        z = Parameter("foo", ParamType.JSON, required=True, nullable=True)
+        assert z.required
+        assert z.nullable
+        assert z.name == "foo"
+        assert z.type is ParamType.JSON
+
     @pytest.mark.parametrize(
         "test",
         (
             ({"data": "yes"}, False),
-            ({"data": None}, True),
             ({"foo": "yes"}, True),
             ({"foo": None, "data": "yes"}, False),
         ),
     )
     def test_invalid_required(self, test):
-        x = Parameter("data", ParamType.STRING, required=True)
+        x = Parameter("data", ParamType.STRING, required=True, nullable=False)
         json, expected = test
         assert x.invalid(json) is expected
```

Comment:
Why did you remove the null-`"data"` scenario from this test?  (I see that you added the scenario to `test_invalid_nullable()`, but that's a different case....)

Reply:
The rule of separation of headaches?